cmd/agent/app/options/options.go

```diff
@@ -206,6 +219,7 @@ func (o *Options) AddFlags(fs *pflag.FlagSet, allControllers []string) {
 	fs.IntVar(&o.ConcurrentWorkSyncs, "concurrent-work-syncs", 5, "The number of Works that are allowed to sync concurrently.")
 	fs.StringSliceVar(&o.ReportSecrets, "report-secrets", []string{"KubeCredentials", "KubeImpersonator"}, "The secrets that are allowed to be reported to the Karmada control plane during registering. Valid values are 'KubeCredentials', 'KubeImpersonator' and 'None'. e.g 'KubeCredentials,KubeImpersonator' or 'None'.")
 	fs.StringVar(&o.MetricsBindAddress, "metrics-bind-address", ":8080", "The TCP address that the controller should bind to for serving prometheus metrics(e.g. 127.0.0.1:8080, :8080). It can be set to \"0\" to disable the metrics serving.")
+	fs.StringVar(&o.HealthProbeBindAddress, "health-probe-bind-address", ":0", "The TCP address that the controller should bind to for serving health probes(e.g. 127.0.0.1:10357, :10357). It can be set to \"0\" or \"\" to disable serving the health probe.")
```

Comment:
```suggestion
	fs.StringVar(&o.HealthProbeBindAddress, "health-probe-bind-address", "0.0.0.0:10357", "The TCP address that the controller should bind to for serving health probes(e.g. 127.0.0.1:10357, :10357). It can be set to \"0\" or \"\" to disable serving the health probe.")
```

Default values should remain the same as before

Reply:
we cannot set it to `0.0.0.0:10357`, cause `--bind-address` and `--secure-port` can not co-exist with `--health-probe-bind-address`. if one set flags like:
```shell
/bin/karmada-agent ----bind-address=0.0.0.0 --secure-port=8443 --health-probe-bind-address=0.0.0.0:10357
```
it will be validate.

Reply:
If users does not set these arguments before or after the upgrade, then the `health-probe-bind-address` used before the upgrade is `0.0.0.0:10357` and after the upgrade becomes `:0`, is this inconsistency acceptable?

Reply:
fixing it :heart:

Reply:
I'm afraid a dynamic port is not acceptable, otherwise, we can't configure the [liveness probe](https://github.com/karmada-io/karmada/blob/bf2a6044a0de3a6219933dbd029e005cfaf7569c/artifacts/agent/karmada-agent.yaml#L40).